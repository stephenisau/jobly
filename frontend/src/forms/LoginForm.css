.card-body {
    opacity: 0.85;
}