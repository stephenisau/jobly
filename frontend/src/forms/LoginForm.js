import React, { Component } from "react";
import "./LoginForm.css";

class LoginForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      username: "",
      password: ""
    }
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  handleChange(evt) {
    this.setState({
      [evt.target.name]: evt.target.value
    })
  }

  async handleSubmit(evt) {
    evt.preventDefault()
    this.props.signIn(this.state);
  }
  render() {
    return (
      <div className="card-body">
        <form onSubmit={this.handleSubmit}>
          <label>Username</label>
          <input 
            name="username"
            value={this.state.username}
            placeholder="Username"
            onChange={this.handleChange}
            className="form-control"
            type="text" />
          <label>Password</label>
          <input
            type="password"
            name="password"
            value={this.state.password}
            placeholder="Password"
            onChange={this.handleChange}
            className="form-control" />
          <button className="btn btn-primary mt-2" type="submit">Submit</button>
        </form>
   
      </div>
    );
  }
}

export default LoginForm;