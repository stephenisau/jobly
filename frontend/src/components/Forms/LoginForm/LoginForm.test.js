import React from 'react';
import LoginForm from './index';
import renderer from "react-test-renderer";
import { shallow } from 'enzyme';

const renderLoginForm = (args) => {
  const defaultProps = {
    currentUser: {}
  }
  const props = {...defaultProps, ...args};
  return shallow(<LoginForm {...props} />)
};


describe("<LoginForm />", () => {

  it("matches snapshot", () => {
    const tree = renderer.create(<LoginForm />);
    expect(tree).toMatchSnapshot();
  });

  it ("renders login form", () => {
    const wrapper = renderLoginForm();
    expect(wrapper.find("form").length).toBe(1);
  })
});