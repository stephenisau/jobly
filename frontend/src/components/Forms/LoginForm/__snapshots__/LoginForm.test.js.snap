// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> matches snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label>
          Username
        </label>
        <input
          className="form-control"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Password
        </label>
        <input
          className="form-control"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <button
        className="btn btn-primary mt-2"
        type="submit"
      >
        Submit
      </button>
      <button
        className="btn btn-info mt-2 ml-2"
        onClick={[Function]}
        type="button"
      >
         Demo 
      </button>
    </form>
  </div>
</div>
`;
